### DISCUSSION TITLE
How to set hooks when using baseapplication

### DISCUSSION THREAD

**Discussion by lmussier** [benoitc/gunicorn#3133] [Q&A]
Hi,

I'm trying to setup a flask app with some shared data using multiprocessing.Manager.
doing so I encounter something similar to #1391 (which is not related to gunicorn), one of the solution pointed in the issue si to use hooks to remove exit handlers.

The issue I have is that I'm not able to get those hooks configured.

```
class StandaloneApplication(gunicorn.app.base.BaseApplication):

    def __init__(self, app, options=None):
        self.options = options or {}
        self.application = app
        super().__init__()

    def load_config(self):
        config = {key: value for key, value in self.options.items()
                  if key in self.cfg.settings and value is not None}
        for key, value in config.items():
            self.cfg.set(key.lower(), value)

    def load(self):
        return self.application

# We check if we are running directly or not
if __name__ == '__main__':

    manager = BaseManager(address=('', 9000), authkey=None)
    manager.start()
    app.logger.info(f'Db listening @ {manager.address}')
    app.logger.info("Starting shared db")

    options = {
        'bind': '%s:%s' % ('127.0.0.1', '9091'),
        'workers': number_of_workers(),
        'config': './gunicorn.conf.py',
    }

    # gunicorn starts here
    StandaloneApplication(app, options).run()
```

Seems to have no effect (to be sure a set conf file to something inexistsant).
Could you point me a way to do this? Either by settting the config file, or by overloading soemthing in base app.
For the reccords I've tried the same code but inheriting from gunicorn.app.base.Application, but got the same result.

Regards.

**Comment by jon-stewart** [comment]
Hi @lmussier, I had the same problem. This code helped me https://github.com/thatmattlove/hyperglass/blob/b5a67e7c0e94b851e0e70bd9f56b27969519840c/hyperglass/main.py#L175

You need to add the function to the dict of options.

```
options["post_worker_init"] = post_worker_init
```
